src/Raven.Server/Rachis/TimeoutEvent.cs

```diff
@@ -134,5 +136,16 @@ public void Dispose()
             DisableTimeout();
             _timer.Dispose();
         }
+
+        public void LowMemory()
+        {
+            // will defer the timeout event if we detected low-memory
+            _timeoutEventSlim.Set();
```

Comment:
Note that this will cause us to defer this once per 5 seconds, so that wouldn't typically do much for us given the typical election timeout